bokehjs/src/coffee/models/plots/plot.coffee

```diff
@@ -352,11 +352,16 @@ class PlotView extends Renderer.View
     @update_range(null)
 
   build_levels: () ->
+    renderer_models = @mget("renderers")
+    for tool_model in @mget("tools")
+      synthetic = tool_model.get("synthetic_renderers") ? []
+      renderer_models = renderer_models.concat(synthetic)
```

Comment:
Some tools already have `renderers` property for a different purpose. So perhaps `computed_renderers`, but that's used as well (at least since this PR). I was considering just adding some property with those renderers to tools, but for some reason (which I can't recall right now) I thought it will be better to set them on a plot. I will give this more thought this week.